packages/turbo-codemod/__tests__/migrate-env-var-dependencies.test.ts

```diff
@@ -0,0 +1,359 @@
+        outputs: [],
+      },
+      dev: {
+        cache: false,
+      },
+      build: {
+        outputs: ["dist/**/*", ".next/**/*"],
+        dependsOn: ["^build", "$TASK_ENV_KEY", "$ANOTHER_ENV_KEY"],
+      },
+    },
+  };
+
+  return merge(config, override, {
+    arrayMerge: (_, sourceArray) => sourceArray,
+  });
+};
+
+describe("migrate-env-var-dependencies", () => {
+  describe("hasLegacyEnvVarDependencies", () => {
+    it("finds env keys in legacy turbo.json - has keys", async () => {
+      const config = getTestTurboConfig();
+      const { hasKeys, envVars } = hasLegacyEnvVarDependencies(config);
+      expect(hasKeys).toEqual(true);
+      expect(envVars).toMatchInlineSnapshot(`
+              Array [
+                "$GLOBAL_ENV_KEY",
+                "$TASK_ENV_KEY",
+                "$ANOTHER_ENV_KEY",
+              ]
+          `);
+    });
+
+    it("finds env keys in legacy turbo.json - multiple pipeline keys", async () => {
+      const config = getTestTurboConfig({
+        pipeline: { test: { dependsOn: ["$MY_ENV"] } },
+      });
+      const { hasKeys, envVars } = hasLegacyEnvVarDependencies(config);
+      expect(hasKeys).toEqual(true);
+      expect(envVars).toMatchInlineSnapshot(`
+              Array [
+                "$GLOBAL_ENV_KEY",
+                "$MY_ENV",
+                "$TASK_ENV_KEY",
+                "$ANOTHER_ENV_KEY",
+              ]
+          `);
+    });
+
+    it("finds env keys in legacy turbo.json - no keys", async () => {
+      // override to exclude keys
+      const config = getTestTurboConfig({
+        globalDependencies: [],
+        pipeline: { build: { dependsOn: [] } },
+      });
+      const { hasKeys, envVars } = hasLegacyEnvVarDependencies(config);
+      expect(hasKeys).toEqual(false);
+      expect(envVars).toMatchInlineSnapshot(`Array []`);
+    });
+  });
+
+  describe("migratePipeline", () => {
+    it("migrates pipeline with env var dependencies", async () => {
+      const config = getTestTurboConfig();
+      const { build } = config.pipeline;
+      const pipeline = migratePipeline(build);
+      expect(pipeline).toHaveProperty("env");
+      expect(pipeline?.env).toMatchInlineSnapshot(`
+        Array [
+          "TASK_ENV_KEY",
+          "ANOTHER_ENV_KEY",
+        ]
+      `);
+      expect(pipeline?.dependsOn).toMatchInlineSnapshot(`
+        Array [
+          "^build",
+        ]
+      `);
+    });
+
+    it("migrates pipeline with no env var dependencies", async () => {
+      const config = getTestTurboConfig();
+      const { test } = config.pipeline;
+      const pipeline = migratePipeline(test);
+      expect(pipeline.env).toBeUndefined();
+      expect(pipeline?.dependsOn).toMatchInlineSnapshot(`
+        Array [
+          "^build",
+        ]
+      `);
+    });
+
+    it("migrates pipeline with existing env key", async () => {
+      const config = getTestTurboConfig({
+        pipeline: { test: { env: ["$MY_ENV"], dependsOn: ["^build"] } },
```

Comment:
I'm not sure we want this behavior in the codemod. AFAIK we're going to hard-error on this from `turbo`. I think we'd prefer to do that rather than edit the user's config, especially if in the future when we can support `$`-prefixed env vars

Reply:
Good call, updated.